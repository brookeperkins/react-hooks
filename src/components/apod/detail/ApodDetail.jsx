import React from 'react';
import PropTypes from 'prop-types';
/* eslint-disable max-len */

const ApodDetail = ({ title, url, explanation }) => (
  <figure>
    <h1>{title}</h1>
    <img src={url} alt={title} width="400" />
    <h3>About: {explanation}</h3>
  </figure>
);

ApodDetail.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
  explanation: PropTypes.string.isRequired
};

export default ApodDetail; 
