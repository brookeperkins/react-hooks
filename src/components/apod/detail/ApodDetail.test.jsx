import React from 'react';
import { render } from '@testing-library/react';
import ApodDetail from './ApodDetail';

describe('ApodDetail component', () => {
  it('renders the ApodDetail component', () => {
    const { asFragment } = render(<ApodDetail 
      title="Messier 76"
      date="1990-01-01"
      url="https://apod.nasa.gov/apod/image/0611/Messier76_seip_smallcc.jpg"
      explanation="Nebula at the right foot of Andromeda..."
    />);
    expect(asFragment()).toMatchSnapshot();
  });
});
