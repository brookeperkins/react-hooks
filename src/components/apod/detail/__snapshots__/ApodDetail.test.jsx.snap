// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ApodDetail component renders the ApodDetail component 1`] = `
<DocumentFragment>
  <figure>
    <h1>
      Messier 76
    </h1>
    <img
      alt="Messier 76"
      src="https://apod.nasa.gov/apod/image/0611/Messier76_seip_smallcc.jpg"
      width="400"
    />
    <h3>
      About: Nebula at the right foot of Andromeda...
    </h3>
  </figure>
</DocumentFragment>
`;
